### PR TITLE
Added an option to change match order

### PR DESCRIPTION
**Pull request by addepar-andy**
Resolves: https://github.com/jakerella/jquery-mockjax/issues/203

Added the global `matchInRegistrationOrder` setting which controls the order in which Mockjax matches requests to handlers.  Also added a section to the documentation explaining what this does.

If this seems like a reasonable change, please let me know, and I'll write tests for this functionality.

### CODE REVIEW

**Comment by jakerella** [comment]
I think this is a great new option! Really appreciate the contribution. Yes, please do add tests... I'm thinking reversing when there are no mocks, 1 mock, and 3 mocks? Basically try to hit a case where you end up with an out of range issue (hopefully not). Also, maybe the test with 3 mocks should have two versions: one with all the same endpoint (all matching) and one where only the 0 and 2 index match? I'll let you decide what's best though.

Also, can you add a note in the README that this setting allows for handler overrides after the initial mocks are set? That should address the original issue.

**Comment by addepar-andy** [comment]
cool, i will come back with tests and the readme adjustments

File: test/test-match-order.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
+(function(qunit, $) {
+	'use strict';
+	
+	var t = qunit.test;
+    
+	/* ----------------------------- */
+	qunit.module( 'Match Order' );
+  /* ----------------------------- */
+
+  function testMatchOrderTest(matchForward, mocks, url, expectedResult) {
```
Comment (jakerella) [inline test/test-match-order.js]: This is a really great abstraction. Well done!

**Comment by jakerella** [comment]
This looks good on the face of things. I'll try to take a closer look this weekend and double check things, then merge and probably cut a minor release.

**Comment by addepar-andy** [comment]
thanks!

**Pull request merged**